### PR TITLE
Demo module example page with usages of QueryModel API

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
With the release of the first public API version of the @labkey/components package, we have some example code in the Public API Docs page of that repo. This PR takes those examples and puts them in use for an example page in the demo module. The example page includes usages of DetailPanelWithModel, GridPanel, and withQueryModels.

#### Changes
* Add new demo module entryPoint with example usages of DetailPanelWithModel, GridPanel, and withQueryModels

### CODE REVIEW

**Comment by cnathe** [comment]
@labkey-nicka @labkey-alan Any chance one of you would have a chance to review this?

File: demo/src/client/QueryModelPage/ExampleGridPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    renderGridButtons = () => {
+        return (
+            <button className={'labkey-button'} onClick={this.onRefreshGrid}>
+                Refresh Grid
+            </button>
+        )
+    };
+
+    render() {
+        const { queryModels, actions, title, asPanel } = this.props;
```
Comment (labkey-susanh) [inline demo/src/client/QueryModelPage/ExampleGridPanel.tsx:37]: Is it worth a comment here that queryModels and actions come from InjectedQueryModels?

File: demo/src/client/QueryModelPage/ExampleGridPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                hideEmptyViewMenu={true}
+            />
+        );
+    }
+}
+
+// Next wrap your component with withQueryModels, here we set the type
+// to OwnProps so the returned component, ExampleGridPanel, can
+// be used in a type safe manner. In this case, if the user forgets to
+// pass in a title we'll get a compiler error as intended.
```
Comment (labkey-susanh) [inline demo/src/client/QueryModelPage/ExampleGridPanel.tsx]: type safe -> type-safe 

"If a user forgets to pass in a title or the asPanel property ...".

**Review by labkey-susanh — APPROVED** [review]

File: demo/src/client/QueryModelPage/ExampleGridPanel.tsx (hunk 1 above)
Comment (cnathe) [reply demo/src/client/QueryModelPage/ExampleGridPanel.tsx:37]: added

File: demo/src/client/QueryModelPage/ExampleGridPanel.tsx (hunk 2 above)
Comment (cnathe) [reply demo/src/client/QueryModelPage/ExampleGridPanel.tsx]: done

**Pull request merged**